packages/alfa-rules/src/sia-r15/rule.ts

```diff
@@ -57,18 +53,24 @@ export default Rule.Atomic.of<Page, Iterable<Element>, Question>({
 
               return groups;
             }, Map.empty<Option<string>, List<Element>>())
-            .values()
+            .values(),
+          (group) => group.size > 1
         );
       },
 
       expectations(target) {
-        const sources = Set.from(
-          map(target, (iframe) => iframe.attribute("src"))
+        const embedSameResource = [...target].every(
+          (element, i, elements) =>
+            // This is either the first element...
+            i === 0 ||
+            // ...or an element that embeds the same resource as the element
+            // before it.
+            referenceSameResource(response.url)(element, elements[i - 1])
```

Comment:
This is closer to what I had in mind: A pairwise comparison that bails out as soon as possible.

Reply:
That's actually computing the embedded resource of each element twice (for comparing with previous and next) while first mapping an embedded resource computation and then comparing them would do it only once (and should also bail out asap if done lazily).
Not really important because this is probably not costly…

Reply:
Nah, it's just comparing some strings and parsing some URLs 🤷